src/main/java/org/fogbowcloud/manager/core/models/linkedlist/SynchronizedDoubleLinkedList.java

```diff
@@ -0,0 +1,108 @@
+package org.fogbowcloud.manager.core.models.linkedlist;
+
+import org.fogbowcloud.manager.core.models.orders.Order;
+
+public class SynchronizedDoubleLinkedList implements ChainedList {
+
+	private Node head;
+	private Node tail;
+	private Node current;
+
+	public SynchronizedDoubleLinkedList() {
+		this.head = this.tail = this.current = null;
+	}
+
+	public synchronized Node getCurrent() {
+		return this.current;
+	}
+
+	protected synchronized Node getHead() {
+		return this.head;
+	}
+
+	protected synchronized Node getTail() {
+		return this.tail;
+	}
+
+	@Override
+	public synchronized void addItem(Order order) {
+		if (order == null) {
+			throw new IllegalArgumentException("Order cannot be null.");
+		}
+		if (this.head == null) {
+			Node firstNode = new Node(null, order, null);
+			this.tail = this.head = this.current = firstNode;
+		} else {
+			Node newItem = new Node(this.tail, order, null);
+			this.tail.setNext(newItem);
+			this.tail = newItem;
+		}
+	}
+
+	@Override
+	public synchronized void resetPointer() {
+		this.current = this.head;
+	}
+
+	@Override
+	public synchronized Order getNext() {
+		if (this.current == null) {
+			return null;
+		}
+		Order orderCurrent = this.current.getOrder();
+		Node nextNode = this.current.getNext();
+		this.current = nextNode;
+		return orderCurrent;
+	}
+
+	/**
+	 * This method removes a given order. Note that this remove method should not
+	 * modify the current pointer (i.e., after removing, the current pointer, must
+	 * point to the same element before this operation), unless the order to be
+	 * removed, is the one in current pointer. In this case, we must remove this
+	 * order, and point current to the next one.
+	 *
+	 * @param order
+	 * @return True if it was removed from the list.
+	 * 		   False, if another thread removed this order from the list and the order couldn't be find.
+	 */
+	@Override
+	public synchronized boolean removeItem(Order order) {
+		if (order == null) {
+			throw new IllegalArgumentException("Order cannot be null.");
+		}
+		Node nodeToRemove = findNodeToRemove(order);
+		if (nodeToRemove == null) {
+			return false;
+		}
+		if (nodeToRemove.getPrevious() != null) {
+			nodeToRemove.getPrevious().setNext(nodeToRemove.getNext());
+		} else { // removing the head
+			this.head = nodeToRemove.getNext();
+		}
+		if (nodeToRemove.getNext() != null) {
+			nodeToRemove.getNext().setPrevious(nodeToRemove.getPrevious());
+		} else { // removing the tail
+			this.tail = nodeToRemove.getPrevious();
+		}
+		if (this.current == nodeToRemove) { // fix current, if current was pointing to cell just removed
+			this.current = nodeToRemove.getNext();
+		}
+		return true;
+	}
+
+	/**
+	 * @param order - Never null
+	 */
+	protected synchronized Node findNodeToRemove(Order order) {
+		Node currentNode = this.head;
+		while (currentNode != null) {
+			if (order.equals(currentNode.getOrder())) {
```

Comment:
We compare the reference to the order, an not the other itself. Instead of equals, use ==.